src/main/java/com/danielagapov/spawn/Mappers/ChatMessageMapper.java

```diff
@@ -13,19 +15,19 @@ public static ChatMessageDTO toDTO(ChatMessage entity) {
         return new ChatMessageDTO(
                 entity.getId(),
                 entity.getTimestamp(),
-                entity.getUserSenderId(),
+                entity.getUserSender().getId(),
                 entity.getContent(),
-                entity.getEventId()
+                entity.getEvent().getId()
         );
     }
 
-    public static ChatMessage toEntity(ChatMessageDTO dto) {
+    public static ChatMessage toEntity(ChatMessageDTO dto, User userSender, Event event) {
```

Comment:
We shouldn't need to pass in extra arguments to these mapper methods.

Reply:
This logic of relations to the event **should be done at the service layer instead.** DTO <-> Entity mappings are just mappings between one DTO to & from its own entity, not inclusive of others.

Reply:
One key advantage is to handle exceptions in the service layer (where they're meant to be), and not including that "responsibility" in these mapper classes.